src/main/java/ir/sahab/kafkaconsumer/OffsetTracker.java

```diff
@@ -245,19 +268,24 @@ boolean ack(int offset) {
                 return false;
             }
 
-            // Set the bit representing this offset.
+            // Clear the bit representing this offset.
             int effectiveOffset = offset - margin;
-            bits.set(effectiveOffset);
-
-            // Find number of consecutive offsets which are acked starting from margin.
-            if (effectiveOffset == numConsecutive) {
-                numConsecutive++;
-                while (bits.get(++effectiveOffset) && numConsecutive < this.effectiveSize)
-                    numConsecutive++;
-            }
+            bits.clear(effectiveOffset);
 
             // Return true if all expected offsets are acked.
-            return (numConsecutive == effectiveSize);
+            return bits.isEmpty() && (maxTrackedOffset == effectiveSize - 1 || noMoreTracks);
```

Comment:
```suggestion
            return isCompleted();
```


Extract isCompleted() method:

bits.isEmpty() && (maxTrackedOffset == effectiveSize - 1 || noMoreTracks);